### ISSUE TITLE
GenerateIncludesExcludes function may have problem when passing exclude *

### ISSUE THREAD

**Issue by qiuming-best** [vmware-tanzu/velero#5020]
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)

if somebody calls the`GenerateIncludesExcludes` function with passing excludes `*`, which means want to exclude all resources. But in this function work out with exclude nothing.  it is not easy to use the wildcard `*` to exclude everything in Kubernetes. Maybe we can return one error with not supporting `*` to avoid of unexpected problems

**What did you expect to happen:**

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`): 
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by sseago** [comment]
Yes -- I recently fixed a bug resulting from velero adding "*" to exclude under certain circumstances. The intention was to "exclude all", but the documentation in the code states that wildcards aren't supported there, so the "*" is ignored. Erroring out on exclude "*" may be better than ignoring, although we'll have to change the method signature to support it -- the GetResourceIncludesExcludes code doesn't currently have an error return, so we'd need to add it (and make sure that everywhere it's called is able to handle an error returned).
